### DISCUSSION TITLE
Using SqlBuilder without Database object.

### DISCUSSION THREAD

**Discussion by DynamicCommander** [maxtoroq/DbExtensions#57] [Q&A] [answered]
I just need SqlBuilder without the database connection part. I am generating a query string but without the use of the Database object it never fully builds the command. Is there a way around this? I'm doing my own connection code already, just ran into the issue of dynamically creating queries.

Thanks!

**Comment by maxtoroq** [comment]
You can pass your existing connection to the Database constructor.

**Comment by DynamicCommander** [comment]
When I try to create a new Database() object, but I'm getting an error that I need a default connection name key in appSettings configuration section. I've looked through the documentation and cannot find anything that references this.

**Comment by maxtoroq** [comment]
You said you already have a connection, then do this:

```csharp
var db = new Database(conn);
```

BTW, what version are you using?

**Comment by DynamicCommander** [comment]
Yes but my connection that I already have isn't using DbExtensions, sorry should have been clear. I'm using System.Data.SqlClient SqlConnection for that. So what I was trying to do was create a Database using the connection string that I have built ""Server=..., ...;Database=...;User ID=...;Password=..;" format. Maybe I'm just really confused about how to use DbExtensions. Basically what I'm trying to use it for is just to build a SQL Query to use in my own stuff, with the connection I have set up.

**Comment by DynamicCommander** [comment]
I'm using this in conjunction with Unity so I'm stuck in .NET3.5 so DbExtensions v4.

**Comment by maxtoroq** [comment]
In v4 you don't need Database, call ToCommand and pass your SqlConnection. See:

https://github.com/maxtoroq/DbExtensions/blob/v4/docs/SqlBuilder.md
https://github.com/maxtoroq/DbExtensions/blob/v4/docs/api/DbExtensions/SqlBuilder/ToCommand.md

**Comment by DynamicCommander** [comment]
Okay this is leading me in the right path, but I'm still getting the an error. "ConfigurationErrorsException: Failed to find or load the registered .Net Framework Data Provider 'System.Data.SqlClient.SqlConnection' System.Data.COmmon.DbProviderFactories.GetFactory(System.String providerInvariantName)

Any ideas? Also, thank you so much for your help, been stuck on this for a while.

**Comment by maxtoroq** [comment, marked as answer]
That's weird. Try passing `SqlClientFactory.Instance` instead.

**Comment by DynamicCommander** [comment]
Okay that worked! Thank you for your help, I appreciate it!
